File: os-console/lib/adapters/learnings.js
// Team learnings adapter — .claude/team-learnings.md (gated: a save is the approval).
import * as gitlib from '../gitlib.js';
import * as md from '../mdparse.js';
import * as repo from '../repo.js';

export const FILE = '.claude/team-learnings.md';
const CAP_LINES = 30; // the file's own stated budget: "hard cap ~30 lines of entries"

export function build() {
  const text = repo.readTextOrNull(FILE);
  if (text === null) return { path: FILE, exists: false, entries: [], capLines: CAP_LINES };
  const body = md.section(text, 'Entries');
  const entries = [];
  for (const b of md.bullets(body)) {
    const m = b.match(/^(\d{4}-\d{2}-\d{2})\s*[—-]+\s*([\s\S]*)$/);
    entries.push(m ? { date: m[1], text: m[2] } : { date: null, text: b });
  }
  return {
    path: FILE,
    exists: true,
    entries,
    seedNote: body.includes('seed examples'),
    entryLines: entries.length,
    capLines: CAP_LINES,
  };
}

export function add(text, settings) {
  const clean = (text || '').trim().replace(/\s+/g, ' ');
  if (clean.length < 8) throw repo.httpErr(400, 'a learning needs at least a sentence');
  if (clean.length > 400) throw repo.httpErr(400, 'keep it to one line (≤400 chars) — that is the format');
  let raw = repo.readText(FILE);
  if (!raw.endsWith('\n')) raw += '\n';
  raw += `- ${md.today()} — ${clean}\n`;
  repo.writeText(FILE, raw);
  const commit = gitlib.commitPaths([FILE], 'console: team learning added');
  const push = gitlib.maybePush(settings);
  return { ...build(), commit, push };
}

File: os-console/pylib/adapters/learnings.py
# Team learnings adapter — .claude/team-learnings.md (gated: a save is the approval).
# Port of lib/adapters/learnings.js — keep the two in lockstep.
import re

from .. import gitlib
from .. import mdparse as md
from .. import repo

FILE = '.claude/team-learnings.md'
CAP_LINES = 30  # the file's own stated budget: "hard cap ~30 lines of entries"


def build():
    text = repo.read_text_or_null(FILE)
    if text is None:
        return {'path': FILE, 'exists': False, 'entries': [], 'capLines': CAP_LINES}
    body = md.section(text, 'Entries')
    entries = []
    for b in md.bullets(body):
        m = re.match(r'^(\d{4}-\d{2}-\d{2})\s*[—-]+\s*(.*)$', b)
        entries.append({'date': m.group(1), 'text': m.group(2)} if m else {'date': None, 'text': b})
    return {
        'path': FILE,
        'exists': True,
        'entries': entries,
        'seedNote': 'seed examples' in body,
        'entryLines': len(entries),
        'capLines': CAP_LINES,
    }


def add(text, settings):
    clean = re.sub(r'\s+', ' ', (text or '').strip())
    if len(clean) < 8:
        raise repo.http_err(400, 'a learning needs at least a sentence')
    if len(clean) > 400:
        raise repo.http_err(400, 'keep it to one line (≤400 chars) — that is the format')
    raw = repo.read_text(FILE)
    if not raw.endswith('\n'):
        raw += '\n'
    raw += '- %s — %s\n' % (md.today(), clean)
    repo.write_text(FILE, raw)
    commit = gitlib.commit_paths([FILE], 'console: team learning added')
    push = gitlib.maybe_push(settings)
    out = build()
    out.update({'commit': commit, 'push': push})
    return out
